sample_consensus/include/pcl/sample_consensus/impl/lmeds.hpp

```diff
@@ -96,22 +97,36 @@ pcl::LeastMedianSquares<PointT>::computeModel (int debug_verbosity_level)
       ++skipped_count;
       continue;
     }
+    // Move all NaNs in distances to the end
+    const auto new_end = (sac_model_->getInputCloud()->is_dense ? distances.end() : std::partition (distances.begin(), distances.end(), [](double d){return !std::isnan (d);}));
+    const auto nr_valid_dists = (sac_model_->getInputCloud()->is_dense ? distances.size() : std::distance (distances.begin (), new_end));
```

Comment:
The check for `is_dense` here is not needed since `new_end` (I'd recommend naming it `number_end` or something) is already `distances.end()` for the computation.

Reply:
Yes, it is not needed, but why call `std::distance` when `distances.size()` already has the desired value?

Reply:
`size` usually calls the difference since storing `front` and `back` pointers is often what the implementations do. See [godbolt](https://godbolt.org/z/pBU-Jm) to see that the codegen is identical. So we can opt for the simpler solution:

```suggestion
    const auto nr_valid_dists = std::distance (distances.begin (), new_end);
```